scripts/dev_setup.sh

```diff
@@ -220,8 +251,12 @@ function install_awscli {
       mkdir -p "$TMPFILE"/work/
       curl -sL -o "$TMPFILE"/aws.zip  "https://awscli.amazonaws.com/awscli-exe-$(uname -s | tr '[:upper:]' '[:lower:]')-${MACHINE}.zip"
       unzip -qq -d "$TMPFILE"/work/ "$TMPFILE"/aws.zip
-      mkdir -p "${HOME}"/.local/
-      "$TMPFILE"/work/aws/install -i "${HOME}"/.local/aws-cli -b "${HOME}"/bin
+      TARGET_DIR="${HOME}"/.local/
+      if [[ "$OPT_DIR" == "true" ]]; then
+         TARGET_DIR="/opt/aws"
+      fi
+      mkdir -p "${TARGET_DIR}" || true
```

Comment:
the script fails on errors with set -eo pipefile, the script needs to be idempotent, runnable on a docker image it has already be run on, where that folder would already exist and mkdir would return an error since it didn't make anything.   Also the script should upgrade older versions on the docker image it was already run on -- just for clarity as to why it's called at all in the build-setup action.

Reply:
`mkdir -p`  does not fail if dir already exists. It only fails if there are access issues or exotic stuff like disk quota or  filename characters not supported by filesystem . In which case all further usage of this dir is likely to fail too. I suspect it just never happened  but if it did, I think the code below would also have failed since there are no dirs. 

It's likely to have no impact because user of this script always has enough rights , but if `|| true` was ever reached, I think the very next lines would have failed anyways, but the troubleshooting would have been slightly harder.